pkg/listening-sockets/listening-sockets_test.go

```diff
@@ -0,0 +1,205 @@
+	LISTEN 0      4096      127.0.0.1:8797  0.0.0.0:* users:(("machine-config-",pid=3534,fd=3))                
+	LISTEN 0      4096      127.0.0.1:8798  0.0.0.0:* users:(("machine-config-",pid=3534,fd=13))               
+	LISTEN 0      4096      127.0.0.1:9100  0.0.0.0:* users:(("node_exporter",pid=4147,fd=3))`
+
+	expectedUDPOutput = `node: test-node
+	UNCONN 0      0           0.0.0.0:111   0.0.0.0:* users:(("rpcbind",pid=1399,fd=5),("systemd",pid=1,fd=78))
+	UNCONN 0      0         127.0.0.1:323   0.0.0.0:* users:(("chronyd",pid=1015,fd=5))                        
+	UNCONN 0      0      10.46.97.104:500   0.0.0.0:* users:(("pluto",pid=2115,fd=21))`
+)
+
+var (
+	clientset *client.ClientSet
+	mockUtils *mock_utils.MockUtilsInterface
+	ctrlTest  *gomock.Controller
+
+	mockPod = &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mock-pod",
+			Namespace: "mock-namespace",
+		},
+	}
+
+	expectedssMat = []types.ComDetails{
+		{
+			Direction: "Ingress",
+			Protocol:  "UDP",
+			Port:      111,
+			NodeRole:  "master",
+			Service:   "rpcbind",
+			Namespace: "",
+			Pod:       "",
+			Container: "test-container",
+			Optional:  false,
+		},
+		{
+			Direction: "Ingress",
+			Protocol:  "UDP",
+			Port:      500,
+			NodeRole:  "master",
+			Service:   "pluto",
+			Namespace: "",
+			Pod:       "",
+			Container: "test-container",
+			Optional:  false,
+		},
+	}
+)
+
+var _ = Describe("GenerateSS", func() {
+	// creating the fake clients, node, pods
+	BeforeEach(func() {
+		sch := runtime.NewScheme()
+
+		err := v1.AddToScheme(sch)
+		Expect(err).NotTo(HaveOccurred())
+
+		testNode := &v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-node",
+				Labels: map[string]string{
+					"node-role.kubernetes.io/master": "",
+				},
+			},
+		}
+
+		fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode).Build()
+		fakeClientset := fakek.NewSimpleClientset()
+
+		clientset = &client.ClientSet{
+			Client:          fakeClient,
+			CoreV1Interface: fakeClientset.CoreV1(),
+		}
+
+		ctrlTest = gomock.NewController(GinkgoT())
+		mockUtils = mock_utils.NewMockUtilsInterface(ctrlTest)
+	})
+
+	DescribeTable("should generate the correct ss tcp, udp output and the correct ssMatrix",
+		func(tc ssTestCase) {
+
+			// RunCommandOnPod had more that one calling and in each call we want other output
+			mockUtils.EXPECT().RunCommandOnPod(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(pod *v1.Pod, cmd []string) ([]byte, error) {
+					switch {
+					case len(cmd) > 2 && strings.HasPrefix(cmd[2], "crictl ps -o json --id"):
+						return []byte(crictlExecCommandOut), nil // container data output
+
+					case strings.HasPrefix(cmd[2], "cat /proc/") &&
+						strings.Contains(cmd[2], "/cgroup"):
+						return []byte(procExecCommandOutput), nil // pid data output
+
+					case cmd[2] == "ss -anpltH":
+						return []byte(tcpExecCommandOutput), nil // tcp output
+
+					case cmd[2] == "ss -anpluH":
```

Comment:
Is it possible to make all these as the others?
mockUtils.EXPECT().
		RunCommandOnPod("ss -anpltH").
		Return([]byte(tcpExecCommandOutput), nil).
		AnyTimes()

Reply:
yah i tried that it will go always to one of them
we are creating fake output from same function so every time im checking what is the cmd that we want to fake